packages/react/src/hooks/accounts/useSigner.ts

```diff
@@ -28,29 +28,34 @@ export const useSigner = ({ skip }: Config = {}) => {
   } = useContext()
   const [state, setState] = React.useState<State>(initialState)
 
+  const cancelQuery = useCancel()
   const getSigner = React.useCallback(async () => {
+    let didCancel = false
+    cancelQuery(() => {
+      didCancel = true
+    })
+
     try {
       setState((x) => ({ ...x, error: undefined, loading: true }))
       const signer = await connector?.getSigner()
-      setState((x) => ({ ...x, data: signer, loading: false }))
+      if (!didCancel) {
+        setState((x) => ({ ...x, data: signer, loading: false }))
+      }
       return signer
     } catch (error_) {
       const error = <Error>error_
-      setState((x) => ({ ...x, data: undefined, error, loading: false }))
+      if (!didCancel) {
+        setState((x) => ({ ...x, data: undefined, error, loading: false }))
+      }
     }
-  }, [connector])
+  }, [cancelQuery, connector])
 
   React.useEffect(() => {
-    if (skip) return
-
-    let didCancel = false
-    if (didCancel) return
-    getSigner()
-
-    return () => {
-      didCancel = true
+    if (!skip && cacheBuster) {
```

Comment:
`cacheBuster` is always truthy since it starts at `1`, this is just to make ESLint happy.